receipt.go

```diff
@@ -68,11 +122,206 @@ type ReceiptFullInfo struct {
 	ReceivedCard  int64  `json:"ReceivedCard"`
 	TotalVAT      int64  `json:"TotalVAT"`
 	ItemsCount    int    `json:"ItemsCount"`
+	ItemsHash     string `json:"ItemsHash"`
 }
 
 type TotalAmount struct {
 	Sale   int64 `json:"Sale"`
 	Refund int64 `json:"Refund"`
 }
 
-///here methods to implement receipt endpoints
+type txIDReq struct {
+	TXID int64 `json:"TXID"`
+}
+
+// GetTXID returns the txID for a sale
+func (r *receipt) GetTXID(ctx context.Context, params SaleParams) (int64, error) {
+	// Prepare the request body
+	bodyBytes, err := json.Marshal(params)
+	if err != nil {
+		return 0, fmt.Errorf("error marshalling body: %s", err.Error())
+	}
+
+	endpoint := r.gateway.FactoryEndpoint(receiptGetTXIDEndpoint)
+	resp, err := r.gateway.HTTPRequest(
+		ctx,
+		endpoint,
+		http.MethodPost,
+		constants.ContentTypeJSON,
+		bodyBytes,
+		nil,
+	)
+	if err != nil {
+		return 0, fmt.Errorf("error creating request: %s", err.Error())
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		errorResp := errorResponse{}
+		if err = json.Unmarshal(resp.Body, &errorResp); err != nil {
+			return 0, fmt.Errorf("error unmarshalling error response: %s responseBody: %s",
+				err.Error(),
+				string(resp.Body),
+			)
+		}
+		return 0, fmt.Errorf("failed to get txID: %s", errorResp.Reason)
+	}
+
+	// Parse the response
+	var txID int64
+	err = json.Unmarshal(resp.Body, &txID)
+	if err != nil {
+		return 0, fmt.Errorf("error unmarshalling response: %s", err.Error())
+	}
+
+	return txID, nil
+}
+
+func (r *receipt) RegisterTXID(ctx context.Context, txID int64) (ReceiptInfo, error) {
+
+	txIDInfo := txIDReq{TXID: txID}
+	reqBody, err := json.Marshal(txIDInfo)
```

Comment:
I think we can omit releasing memory for new object `txIDInfo` and marshal structure without var. And naming here is also faulty in terms of domain logic - `txIDReq` - should be `req` or `request` or even `r` or `txIDRequest` - Info when fast reading confuses

Reply:
fixed